tests/API/Management/ConnectionsTestMocked.php

```diff
@@ -0,0 +1,370 @@
+        ] );
+
+        $response = $api->call()->getAll($strategy);
+
+        $this->assertCount( 3, $response );
+        foreach ($response as $datum) {
+            $this->assertEquals( $strategy, $datum['strategy'] );
+        }
+
+        $this->assertContains( 'strategy='.$strategy, $api->getHistoryQuery() );
+    }
+
+    /**
+     * Test a getAll request with included fields.
+     *
+     * @return void
+     */
+    public function testThatConnectionsGetAllIncludesFields()
+    {
+        $fields = ['id', 'name'];
+        $api    = new MockApi( self::ENDPOINT, [
+            new Response( 200, self::$headers, self::$mockJson->withFields( $fields )->getClean() ),
+            new Response( 200, self::$headers, self::$mockJson->withFields( $fields )->getClean() ),
+        ] );
+
+        $response = $api->call()->getAll(null, $fields);
+
+        $this->assertCount( 10, $response );
+        foreach ($response as $datum) {
+            foreach ($fields as $field) {
+                $this->assertTrue( isset( $datum[$field] ) );
+            }
+        }
+
+        $this->assertContains( 'fields='.implode(',', $fields), $api->getHistoryQuery() );
+
+        // Test an explicit true for includeFields.
+        $api->call()->getAll(null, $fields, true);
+        $this->assertContains( 'include_fields=true', $api->getHistoryQuery(1) );
+    }
+
+    /**
+     * Test a getAll request with excluded fields.
+     *
+     * @return void
+     */
+    public function testThatConnectionsGetAllExcludesFields()
+    {
+        $fields = ['id', 'name'];
+        $api    = new MockApi( self::ENDPOINT, [
+            new Response( 200, self::$headers, self::$mockJson->withFields( $fields, false )->getClean() ),
+        ] );
+
+        $response = $api->call()->getAll(null, $fields, false);
+
+        $this->assertCount( 10, $response );
+        foreach ($response as $datum) {
+            foreach ($fields as $field) {
+                $this->assertFalse( isset( $datum[$field] ) );
+            }
+        }
+
+        $this->assertContains( 'fields='.implode(',', $fields), $api->getHistoryQuery() );
+        $this->assertContains( 'include_fields=false', $api->getHistoryQuery() );
+    }
+
+    /**
+     * Test a paginated getAll request.
+     *
+     * @return void
+     */
+    public function testThatConnectionsGetAllPaginates()
+    {
+        $api = new MockApi( self::ENDPOINT, [
+            new Response( 200, self::$headers, self::$mockJson->withPages( 0, 5 )->getClean() ),
+            new Response( 200, self::$headers, self::$mockJson->withPages( 1, 4 )->getClean() ),
+        ] );
+
+        $response1 = $api->call()->getAll(null, null, null, 0, 5);
+
+        $this->assertContains( 'page=0', $api->getHistoryQuery() );
+        $this->assertContains( 'per_page=5', $api->getHistoryQuery() );
+        $this->assertCount( 5, $response1 );
```

Comment:
what happens when `include_totals=true` is sent in the query? Can your code parse that JSON structure that the server will send as well? Your method should receive the JSON body and return the expected array, or a "page" object if you don't want the user to loose the page totals.

Reply:
> what happens when include_totals=true is sent in the query?

@joshcanhelp you're still missing this use case I think

Reply:
@lbalmaceda - Just passes it straight through as JSON, nothing special to test there.